### PR TITLE
Fixed clippy warnings

### PR DESCRIPTION
**Pull request by Enn3Developer**
Fixed all clippy warnings and tested all the changes.

### CODE REVIEW

File: src/compiler.rs [hunk 1]
```diff
@@ -247,7 +264,12 @@ pub fn compile_tokens(scope: usize, ctokens: Expression) -> String {
 	while let Some(t) = ctokens.next() {
 		result += &match t {
 			SYMBOL(lexeme) => lexeme,
-			VARIABLE {local, names, values, line} => {
+			VARIABLE {
+				local,
+				names,
+				values,
+				line,
+			} => {
```
Comment (Maiori44) [inline src/compiler.rs:272]: is this necessary? it looks kinda ugly to me

**Review by Maiori44 — COMMENTED** [review]
this seems fine, but I wonder what changed from the last time a similar pr was made that doesn't make it break the syntax this time

**Comment by Enn3Developer** [comment]
I was thinking the same, finally I could find the culprit but nada

File: src/compiler.rs (hunk 1 above)
Comment (Enn3Developer) [reply src/compiler.rs:272]: I think that's from the autoformatter

Comment (Maiori44) [reply src/compiler.rs:272]: I'd understand if it tried to make it not go past screen border, but it didnt go past before so it's weird lol

**Comment by Maiori44** [comment]
Maybe you can try comparing the 2 prs

**Pull request merged**